lib/FundDistribution/FundDistributionFields/FundDistributionFieldsFinalContainer.js

```diff
@@ -27,35 +29,62 @@ const FundDistributionFieldsFinalContainer = ({
   stripes,
   totalAmount,
   validate,
+  formValues,
   validateFundDistributionTotal,
 }) => {
   // eslint-disable-next-line react-hooks/exhaustive-deps
+
   const mutator = useMemo(() => originMutator, []);
   const funds = resources?.[DICT_FUNDS]?.records;
   const fundCodes = useMemo(() => funds && new Map(funds.map(f => [f.id, f.code])), [funds]);
   const [expenseClassesByFundId, setExpenseClassesByFundId] = useState({});
 
   useEffect(() => {
-    const expenseClassesFundsToFetch = fundDistribution?.filter(({ fundId }) => (
-      fundId && !expenseClassesByFundId[fundId]
-    ));
-
-    if (expenseClassesFundsToFetch?.length) {
-      expenseClassesFundsToFetch.forEach(({ fundId }) => fetchExpenseClasses(
-        fundId, expenseClassesByFundId, setExpenseClassesByFundId, mutator.fundExpenseClasses,
-      ));
+    const uniqueFundDistribution = uniqBy(fundDistribution, 'fundId');
+
+    uniqueFundDistribution.forEach(({ fundId }) => fetchExpenseClasses({
+      fundId,
+      fiscalYearId,
+      setExpenseClassesByFundId,
+      mutator: mutator.fundExpenseClasses,
+    }));
+  }, [fundDistribution, fiscalYearId, mutator.fundExpenseClasses]);
+
+  const resetExpenseClasses = useCallback(() => {
+    formValues?.adjustments?.forEach(({ fundDistributions }, index) => {
+      // reset form field state if fundId is changed and there is no expense class for this fund
+      fundDistributions?.forEach(({ fundId, expenseClassId }, distributionIndex) => {
+        const fieldName = `adjustments[${index}].fundDistributions[${distributionIndex}]`;
+
+        if (!expenseClassId) return;
+        const isExpenseClassExist = expenseClassesByFundId[fundId]?.find(({ id }) => id === expenseClassId);
+
+        if (!isExpenseClassExist) {
+          change(`${fieldName}.expenseClassId`, null);
+        }
+      });
+    });
+  }, [change, expenseClassesByFundId, formValues?.adjustments]);
+
+  useEffect(() => {
+    // prevent reset expense class on edit if there is selected expense class on initial load
+    if (Object.values(expenseClassesByFundId).length > 0) {
```

Comment:
So, for instance if I have selected 'Previous' expense class from budget 2021 and changed the fiscal year from FY2021 to FY2023, that doesn't have 'Previous' expense class in the budget, `expenseClassId` will be still keep ID of the 'Previous' expense class from budget 2021?